script.js

```diff
@@ -1,56 +1,71 @@
-
-  bookCollection.forEach((book, index) => {
-    const li = document.createElement('li');
-    li.textContent = `${book.title} - ${book.author}`;
-
-    // Function to remove a book from the collection
-
-    function removeBook(index) {
-      bookCollection.splice(index, 1);
-      localStorage.setItem('bookCollection', JSON.stringify(bookCollection));
-      renderBookList();
-    }
-
-    const removeButton = document.createElement('button');
-    removeButton.textContent = 'Remove';
-    removeButton.addEventListener('click', () => {
-      removeBook(index);
-    });
+class BookCollection {
+  constructor() {
+    this.bookListElement = document.getElementById('bookList');
+    this.addBookForm = document.getElementById('addBookForm');
+    this.titleInput = document.getElementById('title');
+    this.authorInput = document.getElementById('author');
+    this.bookCollection = [];
+    this.loadFromLocalStorage();
+    this.addEventListeners();
+    this.renderBookList();
+  }
 
-    li.appendChild(removeButton);
-    bookListElement.appendChild(li);
-  });
-}
+  loadFromLocalStorage() {
+    const savedCollection = localStorage.getItem('bookCollection');
+    this.bookCollection = savedCollection
+      ? JSON.parse(savedCollection)
+      : [];
+  }
 
-// Function to add a new book to the collection
-function addBook(title, author) {
-  const newBook = { title, author };
-  bookCollection.push(newBook);
-  localStorage.setItem('bookCollection', JSON.stringify(bookCollection));
-  renderBookList();
-}
+  saveToLocalStorage() {
+    localStorage.setItem(
+      'bookCollection',
+      JSON.stringify(this.bookCollection),
+    );
+  }
+
+  removeBook(index) {
+    this.bookCollection.splice(index, 1);
+    this.saveToLocalStorage();
+    this.renderBookList();
+  }
 
-// Event handler for the form submission
-document.getElementById('addBookForm').addEventListener('submit', (event) => {
-  event.preventDefault();
-  const titleInput = document.getElementById('title');
-  const authorInput = document.getElementById('author');
-  const title = titleInput.value;
-  const author = authorInput.value;
+  renderBookList() {
+    this.bookListElement.innerHTML = '';
+    this.bookCollection.forEach((book, index) => {
+      const li = document.createElement('li');
+      li.className = 'single-book';
+      li.textContent = `${book.title} - ${book.author}`;
 
-  addBook(title, author);
+      const removeButton = document.createElement('button');
+      removeButton.textContent = 'Remove';
+      removeButton.className = 'remove-btn';
+      removeButton.addEventListener('click', () => {
+        this.removeBook(index);
+      });
 
-  // Reset form inputs
-  titleInput.value = '';
-  authorInput.value = '';
-});
+      li.appendChild(removeButton);
+      this.bookListElement.appendChild(li);
+    });
+  }
+
+  addBook(title, author) {
+    const newBook = { title, author };
+    this.bookCollection.push(newBook);
```

Comment:
[OPTIONAL] I suggest you favor the use of array methods that don't mutate the array😉. you can make use of `concact` here instead😉.